flink-clients/src/main/java/org/apache/flink/client/program/Client.java

```diff
@@ -420,6 +428,60 @@ public JobSubmissionResult run(JobGraph jobGraph, boolean wait) throws ProgramIn
 		}
 	}
 
+
+	/**
+	 * Executes the CANCEL action.
+	 *
+	 * @param args Command line arguments for the cancel action.
+	 */
+
+
+	protected int cancel(JobID jobId){
+		LOG.info("Executing 'cancel' command.");
+
+		final FiniteDuration timeout = AkkaUtils.getTimeout(configuration);
+
+		try {
+			String address = configuration.getString(ConfigConstants.JOB_MANAGER_IPC_ADDRESS_KEY, null);
+			Option<Tuple2<String, Object>> remoting =
+					new Some<Tuple2<String, Object>>(new Tuple2<String, Object>("", 0));
+
+			// start a remote actor system to listen on an arbitrary port
+			ActorSystem actorSystem = AkkaUtils.createActorSystem(configuration, remoting);
+
+
+			ActorRef jobManager = JobManager.getJobManagerRemoteReference(address, actorSystem, timeout);
+			Future<Object> response = Patterns.ask(jobManager, new JobManagerMessages.CancelJob(jobId), new Timeout(AkkaUtils.INF_TIMEOUT()));
+
+			try {
+				Await.result(response, timeout);
+				return 0;
+			}
+			catch (Exception e) {
+				throw new Exception("Canceling the job with ID " + jobId + " failed.", e);
+			}
+		}
+		catch (Throwable t) {
+			return handleError(t);
+		}
```

Comment:
I would handle the error explicitly here and print a stacktrace and info message. Creating a new method is not necessary because it is not used elsewhere. You might want to return a boolean which returns true if the cancellation succeeded (CancellationSuccess received) or failed (otherwise).